website/addons/onedrive/model.py

```diff
@@ -0,0 +1,258 @@
+# -*- coding: utf-8 -*-
+import logging
+import requests
+
+from flask import abort, request
+from datetime import datetime
+
+from modularodm import fields
+
+from framework.auth import Auth
+from framework.exceptions import HTTPError
+
+from website.addons.base import exceptions
+from website.addons.base import AddonOAuthUserSettingsBase, AddonOAuthNodeSettingsBase
+from website.addons.base import StorageAddonBase
+
+from website.addons.onedrive import settings
+from website.addons.onedrive.utils import OneDriveNodeLogger
+from website.addons.onedrive.serializer import OneDriveSerializer
+from website.addons.onedrive.client import OneDriveAuthClient
+from website.addons.onedrive.client import OneDriveClient
+
+from website.oauth.models import ExternalProvider
+
+logger = logging.getLogger(__name__)
+
+logging.getLogger('onedrive1').setLevel(logging.WARNING)
+
+
+class OneDrive(ExternalProvider):
+    name = 'onedrive'
+    short_name = 'onedrive'
+
+    client_id = settings.ONEDRIVE_KEY
+    client_secret = settings.ONEDRIVE_SECRET
+
+    auth_url_base = settings.ONEDRIVE_OAUTH_AUTH_ENDPOINT
+    callback_url = settings.ONEDRIVE_OAUTH_TOKEN_ENDPOINT
+    auto_refresh_url = settings.ONEDRIVE_OAUTH_TOKEN_ENDPOINT
+    default_scopes = ['wl.basic wl.signin onedrive.readwrite wl.offline_access']
+
+    _auth_client = OneDriveAuthClient()
+    _drive_client = OneDriveClient()
+
+    def handle_callback(self, response):        
+        """View called when the Oauth flow is completed. Adds a new OneDriveUserSettings
+        record to the user and saves the user's access token and account info.
+        """
+        
+        userInfoRequest = requests.get(("{}me?access_token={}").format(settings.MSLIVE_API_URL, response['access_token']))
```

Comment:
This confused me at first, but I actually really like having this request made here. :+1: 

I see a method in OneDriveClient that looks like it could handle this call. Is there any reason not to use the client here?